hail/src/main/scala/is/hail/expr/ir/Interpret.scala

```diff
@@ -777,6 +778,89 @@ object Interpret {
         function.execute(ctx, child.execute(ctx))
       case TableAggregate(child, query) =>
         val value = child.execute(ctx)
+
+        val globalsBc = value.globals.broadcast
+        val globalsOffset = value.globals.value.offset
+
+        if (HailContext.getFlag("newaggs") != null) {
+          try {
+            val res = genUID()
+            val extracted = agg.Extract(query, res)
+
+            val wrapped = if (extracted.aggs.isEmpty) {
+              val (rt: PTuple, f) = Compile[Long, Long](
+                "global", value.globals.t,
+                MakeTuple(FastSeq(extracted.postAggIR)))
+
+              Region.scoped { region =>
+                SafeRow(rt, region, f(0, region)(region, globalsOffset, false))
+              }
+            } else {
+              val spec = CodecSpec.defaultUncompressed
+
+              val (_, initOp) = CompileWithAggregators2[Long, Unit](
+                extracted.aggs,
+                "global", value.globals.t,
+                Begin(FastIndexedSeq(extracted.init,
+                  SerializeAggs(0, 0, spec, extracted.aggs))))
+
+              val (_, partitionOpSeq) = CompileWithAggregators2[Long, Long, Unit](
+                extracted.aggs,
+                "global", value.globals.t,
+                "row", value.rvd.rowPType,
+                extracted.seqPerElt)
+
+              val read = extracted.deserialize(spec)
+              val write = extracted.serialize(spec)
+              val combOpF = extracted.combOpF(spec)
+
+              val (rTyp: PTuple, f) = CompileWithAggregators2[Long, Long](
+                extracted.aggs,
+                "global", value.globals.t,
+                Let(res, extracted.results, MakeTuple(FastSeq(extracted.postAggIR))))
+              assert(rTyp.types(0).virtualType == query.typ)
+
+              val initBytes = Region.scoped { region =>
+                val initF = initOp(0, region)
+                Region.scoped { aggRegion =>
+                  initF.newAggState(aggRegion)
+                  initF(region, globalsOffset, false)
+                  initF.getSerializedAgg(0)
+                }
+              }
+
+              val aggResults = value.rvd.combine[Array[Byte]](initBytes,
+                { (i: Int, ctx: RVDContext, it: Iterator[RegionValue]) =>
+                  val partRegion = ctx.freshRegion
+                  val globalsOffset = globalsBc.value.readRegionValue(partRegion)
+                  val seqOps = partitionOpSeq(i, partRegion)
+
+                  Region.smallScoped { aggRegion =>
+                    seqOps.setAggState(aggRegion, read(aggRegion, initBytes))
```

Comment:
Do you think it's better to send/deserialize the initial agg state per partition than to run initF once per partition?

Reply:
It probably depends on the specific aggregators being run, but in the vast majority of cases, probably not. I'll change it.